src/cadence/contracts/AlchemyMetadataWrapperEmulator.cdc

```diff
@@ -0,0 +1,241 @@
+import NonFungibleToken from "./NonFungibleToken.cdc"
+import MetadataViews from "./MetadataViews.cdc"
+
+// $BEGIN NFT addresses
+import TestNFT from "./TestNFT.cdc"
+import TestNFTWithViews from "./TestNFTWithViews.cdc"
+// $END NFT addresses
+
+/*
+    A wrapper contract around the script provided by the Alchemy GitHub respository.
+    Allows for on-chain storage of NFT Metadata, allowing consumers to query upon.
+    This contract will be periodically updated based on new onboarding PRs and deployed.
+    Any consumers calling the public methods below will retrieve the latest and greatest data.
+*/
+pub contract AlchemyMetadataWrapperEmulator {
```

Comment:
We may want a common interface that these three need to adhere to with the Alchemy backwards compatibility spec

Reply:
Testnet and mainnet actually have different struct names - I'm not sure if we need to preserve that difference or not though. Might be worth looking into

Reply:
I think adding templating here mitigates the need for this, take a look and let me know if you agree/disagree.